src/zkstar.rs

```diff
@@ -0,0 +1,118 @@
+use crate::errors::InternalError;
+use libpaillier::unknown_order::BigNumber;
+use serde::{Deserialize, Serialize};
+
```

Comment:
todo: Add module-level docs here, please! We have been inconsistent about this in the rest of the library, but I like to have at least one sentence up here to show up in the built docs.

Reply:
Im not sure I understand what you are asking for here. Can you refer me to an example of what you'd like to see ?

Reply:
You can try building / looking at the docs for [zkchannels-crypto](https://github.com/boltlabs-inc/libzkchannels-crypto/tree/main/zkchannels-crypto) -- the pedersen, pointcheval_sanders, and proofs modules are all good examples.

Generally speaking, I like to have one sentence that's a quick summary for anyone scanning the modules when they build the docs, and then a few more sentences with any particular context that's useful for someone coming into the library. E.g. in this case, I vetoed all your nice suggestions about adding general-purpose functionality (like a method to make a random builder) because I don't think we need it in this crate and I don't want to get confused later about why it's there. But clearly that's something any reasonable cryptographer might notice is missing when they look at the module, so I'd explain in the docs that this is _just_ the application-specific tools we need for this library.

Depending on how complex the module is, I might also add a few comments about expected usage and any major design decisions we made. For example, here we might explain the point of the builder and how it enforces correct construction and prevents silent failures from trying to combine operands of incompatible groups together. But I don't think we need to do that for this module at this point.

I might write it like:
```
//! Utilities for creating and using elements of the multiplicative group of integers mod `N`.
//!
//! This module is designed for protocol-specific use and doesn't capture all the 
//! possible functionalities on group elements.
```